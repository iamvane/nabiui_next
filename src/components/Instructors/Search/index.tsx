import React from 'react'
import Title from "./Title";
import SearchSection1 from "./SearchSection1";
import SearchSection2 from "./SearchSection2";
import Subtitle from "./Subtitle";
import Results from "./Results";

const index = () => {
    return (
        <div>
            <Title />
            <SearchSection1 />
            <SearchSection2 />
            <Subtitle />
            <Results />
        </div>
    )
}


export default index