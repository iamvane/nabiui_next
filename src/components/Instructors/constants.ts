export enum Gender {
  any = 'Any',
  female = 'Female',
  male = 'Male'
}

export enum SortBy {
  Any = 'Any',
  lessonRate = 'Lesson rate',
  Distance = 'Distance',
}

export interface SortByOptions {
  label: SortBy;
  value: string;
}

export const sortByOptions = [
  {
    label: SortBy.Any,
    value: ''
  },
  {
    label: SortBy.lessonRate,
    value: 'rate'
  },
  {
    label: SortBy.Distance,
    value: 'distance'
  }
];

export interface SelectOption {
  label: string;
  value: string;
}

export const ageLabels = {
  children: 'Child (12 & under)',
  teens: 'Teen (13-17)',
  adults: 'Adult (18-65)',
  seniors: 'Senior (65+)'
};

export const ageOptions: SelectOption[] = [
  {
    label: 'Any',
    value: ''
  },
  {
    label: ageLabels.children,
    value: 'children'
  },
  {
    label: ageLabels.teens,
    value: 'teens'
  },
  {
    label: ageLabels.adults,
    value: 'adults'
  },
  {
    label: ageLabels.seniors,
    value: 'seniors'
  },
];

export const genderOptions: SelectOption[] = [
  {
    label: 'Any',
    value: '',
  },
  {
    label: 'Female',
    value: 'female'
  },
  {
    label: 'Male',
    value: 'male'
  }
];

export const availabilityLabels = {
  monday: 'Monday',
  tuesday: 'Tuesday',
  wednesday: 'Wednesday',
  thursday: 'Thursday',
  friday: 'Friday',
  saturday: 'Saturday',
  sunday: 'Sunday'
};

export const availabilityOptions: SelectOption[] = [
  {
    label: availabilityLabels.monday,
    value: 'monday'
  },
  {
    label: availabilityLabels.tuesday,
    value: 'tuesday'
  },
  {
    label: availabilityLabels.wednesday,
    value: 'wednesday'
  },
  {
    label: availabilityLabels.thursday,
    value: 'thursday'
  },
  {
    label: availabilityLabels.friday,
    value: 'friday'
  },
  {
    label: availabilityLabels.saturday,
    value: 'saturday'
  },
  {
    label: availabilityLabels.sunday,
    value: 'sunday'
  },
];

export const placeForLessonsLabels = {
  home: 'Home',
  studio: 'Studio',
  online: 'Online'
};

export const placeForLessonsOptions: SelectOption[] = [
  {
    label: placeForLessonsLabels.home,
    value: 'home'
  },
  {
    label: placeForLessonsLabels.studio,
    value: 'studio'
  },
  {
    label: placeForLessonsLabels.online,
    value: 'online'
  },
];

export const qualificationOptions: SelectOption[] = [
  {
    label: 'Certified teacher',
    value: 'certifiedTeacher'
  },
  {
    label: 'Music therapy',
    value: 'musicTherapy'
  },
  {
    label: 'Music production',
    value: 'musicProduction'
  },
  {
    label: 'Ear training',
    value: 'earTraining'
  },
  {
    label: 'Conducting',
    value: 'conducting'
  },
  {
    label: 'Virtuoso recognition',
    value: 'virtuosoRecognition'
  },
  {
    label: 'Performance',
    value: 'performance'
  },
  {
    label: 'Music theory',
    value: 'musicTheory'
  },
  {
    label: 'Experience teaching young children',
    value: 'experienceTeachingYoungChildren'
  },
  {
    label: 'Repertoire selection',
    value: 'repertoireSelection'
  },
];

export namespace InstructorsComponent {
  export const pageTitle = 'Instructors';
  export const breadcrumbLabels = {
    home: 'Home',
    requests: 'Instructors'
  }
}

export namespace InstructorCardComponent {
  export const maxBioDescriptionLength = 112;

  export enum Text {
    Age =  'Age',
    StartAt = 'Rate per lesson',
    ViewMore = '...View More',
    ViewLess = 'View Less',
    LessonRatePlaceholder = '{LessonRate}',
    LessonRate = '${LessonRate}',
    LastLogin = 'Last Login:',
    LessonsTaught = 'Lessons Taught:',
    MemberSince = 'Member Since:',
    SendMessage = 'Send Message',
    ViewProfile = 'View Profile'
  }
}

export namespace InstructorFilterComponent {
  export const title = 'Refine Your Search';
  export const rangeDefaulValue = [0, 200];

  export const rangeMinValue = 0; // Lessons Rate minimun value
  export const rangeMaxValue = 200; // Lessons Rate maximun value
  export const sliderMarks = [
    {
      value: 0,
      label: '$0',
    },
    {
      value: 200,
      label: '$200',
    },
  ];

  export enum FieldLabels {
    Sort = 'Sort',
    Distance = 'Distance',
    Location = 'Place For Lessons',
    Availability = 'Availability',
    Price = 'Price',
    Age = 'Student Age',
    Gender = 'Gender',
    Qualifications = 'Additional Qualifications'
  }

  export enum FieldNames {
    Sort = 'sort',
    Distance = 'distance',
    Location = 'placeForLessons',
    Availability = 'availability',
    Instrument = 'instrument',
    ZipCode = 'zipCode',
    Age = 'age',
    Gender = 'gender',
    Price = 'lessonRate',
    Qualifications = 'qualifications'
  }

  export enum Ids {
    Sort= 'sort',
    Distance = 'distance',
    Location = 'placeForLessons',
    Availability = 'availability',
    Age = 'age',
    Gender = 'gender',
    ZipCode = 'zipCode',
  }

  export enum Placeholders {
    Sort = 'Sort by',
    Distance = 'Any',
    Location = 'Select location',
    Availability = 'Select day(s)',
    Price = 'Select price',
    Age = 'Select age',
    Gender = 'Select gender',
    Qualifications = 'Select qualifications'
  }
}

export enum Route {
  Profile = '/profile'
}

export interface QualificationsFilterType {
  certifiedTeacher?: boolean;
  musicTherapy?: boolean;
  musicProduction?: boolean;
  earTraining?: boolean;
  conducting?: boolean;
  virtuosoRecognition?: boolean;
  performance?: boolean;
  musicTheory?: boolean;
  youngChildrenExperience?: boolean;
  repertoireSelection?: boolean;
}

export interface PlaceForLessonsFilterType {
  home?: boolean;
  studio?: boolean;
  online?: boolean;
}

export interface InstructorListQueryParams {
  sort?: string;
  instruments?: string;
  location?: string;
  distance?: number;
  placeForLessons?: string;
  availability?: string;
  age?: string;
  gender?: string;
  qualifications?: string;
  min_rate?: number;
  max_rate?: number;
}

export const defaultQueryParams: InstructorListQueryParams = {
};

export const defaultQualifications: QualificationsFilterType = {
};

export const defaultPlaceForLessons: PlaceForLessonsFilterType = {
};
