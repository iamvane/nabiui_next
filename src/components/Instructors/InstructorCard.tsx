import * as React from "react";
import moment from "moment";
const reactStringReplace = require("react-string-replace");
import Router from "next/router";

import {
  Avatar,
  Button,
  Grid,
  IconButton,
  Typography
} from "@material-ui/core";
import Star from "@material-ui/icons/Star";

import { Rates } from "../../redux/models/InstructorModel";
import PlaceForlessons from "../PlaceForLessons/PlaceForLessons";
import { BackgroundCheckStatus } from "../ProfileBuilder/constants";
import { ProfileHeaderComponent } from "../Profile/constants";
import { InstructorCardComponent } from "./constants";

interface Props {
  id: number;
  age: number;
  displayName: string;
  bioTitle: string;
  bioDescription: string;
  reviewsNumber: number;
  rateStartAt: Rates;
  instruments: string[];
  lessonsTaught: number;
  lastLogin: string;
  backgroundCheckStatus: string;
  distance: number;
  experience: number;
  favorite: boolean;
  memberSince: string;
  avatarImage: string;
  placeForLessons: {
    studio?: boolean;
    online?: boolean;
    home?: boolean;
  };
  fetchInstructor: (id: number) => void;
}

const displayRatingStars = (reviewsNumber: number) => {
  let ratingStars: JSX.Element[] = [];
  for (let i = 0; i < 5; i++) {
    if (i < reviewsNumber) {
      ratingStars.push(<Star key={i} />);
    } else {
      ratingStars.push(<Star color="disabled" key={i} />);
    }
  }
  return ratingStars;
};

const selectStartingRate = (rates: Rates) => {
  const rateClasses = ["mins30", "mins45", "min60", "mins90"];
  let selectedRate = [] as number[];
  rateClasses.forEach(rate => {
    if (rates[rate]) {
      selectedRate.push(Number(rates[rate]));
    }
  });
  return selectedRate.length
    ? Math.min(...selectedRate)
        .toFixed(2)
        .toString()
    : "";
};

const InstructorCard: React.StatelessComponent<Props> = props => {
  const BackgroundCheckIcon =
    "https://nabimusic.s3.us-east-2.amazonaws.com/assets/images/nabi-background-check.svg";
  const AvatarStyles = { width: "120px", height: "120px" };
  const {
    age,
    lastLogin,
    lessonsTaught,
    bioDescription,
    bioTitle,
    avatarImage,
    displayName,
    memberSince,
    backgroundCheckStatus,
    instruments,
    rateStartAt,
    id,
    experience,
    distance,
    placeForLessons
  } = props;
  let instrumentItem: string[] = [];
  instruments.map(instrument => {
    instrumentItem.push(
      (instrument as String).charAt(0).toUpperCase() + instrument.slice(1)
    );
    return instrumentItem;
  });
  const navigateToProfile = () => {
    Router.push(`/profile/${id}`);
  };
  return (
    <div onClick={navigateToProfile} className="nabi-section nabi-padding-top-small nabi-padding-bottom-small nabi-background-white nabi-margin-bottom-small nabi-position-relative nabi-cursor-pointer item-card">
      <Grid container={true} spacing={3}>
        <Grid item={true} xs={12} md={3} className="nabi-text-center">
          <Avatar
            src={avatarImage}
            style={AvatarStyles}
            className="nabi-margin-center"
          />
          <Typography className="nabi-margin-top-xsmall nabi-text-semibold">
            {displayName}
          </Typography>
          <div className="nabi-cursor-pointer nabi-color-nabi">
            {displayRatingStars(props.reviewsNumber)}
          </div>
          <Typography className="nabi-text-uppercase">
            {experience} {ProfileHeaderComponent.Text.YearExperiece} | {age}{" "}
            {ProfileHeaderComponent.Text.YearOld}
          </Typography>
          <Typography>
            {distance && String(distance.toFixed(1)) + " mi away"}
          </Typography>
        </Grid>
        <Grid item={true} xs={12} md={6}>
          <p className="nabi-jennasue-title nabi-color-nabi nabi-margin-top-zero nabi-text-center nabi-margin-bottom-xsmall nabi-text-left-md">
            {bioTitle}
          </p>
          <div>
            <Typography className="nabi-margin-top-xsmall">
              {bioDescription &&
                `${bioDescription.slice(
                  0,
                  InstructorCardComponent.maxBioDescriptionLength
                )}...`}
            </Typography>
            <Grid container={true} spacing={1}>
              {backgroundCheckStatus === BackgroundCheckStatus.verified && (
                <Grid item={true} xs={12} className="nabi-margin-top-xsmall">
                  <IconButton
                    color="secondary"
                    className="nabi-display-inline-block"
                  >
                    <img
                      src={BackgroundCheckIcon}
                      className="nabi-custom-button-icon"
                      alt="background-check"
                    />
                  </IconButton>
                  <Typography className="nabi-margin-left-xsmall nabi-display-inline-block">
                    Background Check
                  </Typography>
                </Grid>
              )}
              <Grid item={true} md={6}>
                <Typography>
                  <span className="nabi-text-semibold nabi-color-nabi">
                    {" "}
                    Teaches:{" "}
                  </span>
                  {instrumentItem.join(" ")}
                </Typography>
              </Grid>
              <Grid item={true} md={6}>
                <Typography color="primary" className="nabi-text-semibold">
                  Teaching locations:
                </Typography>
                <PlaceForlessons
                  online={placeForLessons.online}
                  studio={placeForLessons.studio}
                  home={placeForLessons.home}
                />
              </Grid>
            </Grid>
          </div>
        </Grid>
        <Grid item={true} xs={12} md={3} className="nabi-text-center">
          <Typography className="nabi-text-mediumbold">
            {InstructorCardComponent.Text.StartAt}
          </Typography>
          <Typography className="nabi-text-mediumbold">
            {reactStringReplace(
              InstructorCardComponent.Text.LessonRate,
              InstructorCardComponent.Text.LessonRatePlaceholder,
              (i: string) => (
                <span key={i} className="nabi-font-large">
                  ${selectStartingRate(rateStartAt)}
                </span>
              )
            )}
          </Typography>
          <Button
            color="primary"
            className="nabi-responsive-button nabi-margin-top-xsmall"
            variant="contained"
          >
            {InstructorCardComponent.Text.ViewProfile}
          </Button>
          <div className="nabi-margin-top-small">
            <Typography className="nabi-text-mediumbold nabi-margin-right-xsmall nabi-display-inline-block">
              {InstructorCardComponent.Text.LessonsTaught}
            </Typography>
            <Typography className="nabi-display-inline-block">
              {lessonsTaught}
            </Typography>
            <br />
            <Typography className="nabi-text-mediumbold nabi-margin-right-xsmall nabi-display-inline-block">
              {InstructorCardComponent.Text.LastLogin}
            </Typography>
            <Typography className="nabi-display-inline-block">
              {moment(lastLogin).fromNow()}
            </Typography>
            <br />
            <Typography className="nabi-text-mediumbold nabi-margin-right-xsmall nabi-display-inline-block">
              {InstructorCardComponent.Text.MemberSince}
            </Typography>
            <Typography className="nabi-display-inline-block">
              {memberSince}
            </Typography>
          </div>
        </Grid>
      </Grid>
    </div>
  );
};

export default InstructorCard;
