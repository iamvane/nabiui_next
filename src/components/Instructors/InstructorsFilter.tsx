import * as React from 'react';

import {
  Checkbox,
  FormControl,
  Grid,
  Input,
  InputLabel,
  ListItemText,
  MenuItem,
  Select,
  Slider
} from '@material-ui/core';

import { optionsText, placeholder } from 'components/common/constants/DistanceSelect';
import { qualificationTypes } from 'components/Qualifications/constants';

import {
  SelectOption,
  sortByOptions,
  InstructorFilterComponent,
  availabilityLabels,
  availabilityOptions,
  qualificationOptions,
  ageOptions,
  genderOptions,
  placeForLessonsLabels,
  placeForLessonsOptions
} from 'components/Instructors/constants';

interface Props {
  sortBy: string;
  distance: number;
  placeForLessons: string[];
  availability: string[];
  priceRange: number[];
  age: string;
  gender: string;
  qualifications: string[];
  handleChange: (event: React.FormEvent<{}>) => void;
  handlePriceChange: (event: any, newValue: number | number[]) => void;
  handlePriceCommitted: (event: any, newValue: number | number[]) => void;
}

const InstructorFilter: React.StatelessComponent<Props> = props => {
  const milesSelectItems = (): JSX.Element[] => {
    const items = [];
    let index;
    for (index = 1; index < 50; index++) {
      if (index > 5) {
        index += 4;
      }
      items.push(<option key={index} value={index}>{index} {optionsText}</option>);
    }
    return items;
  };

  const selectOptions = (options: SelectOption[]) => {
    return  options.map(item =>
      <option key={item.value} value={item.value}>{item.label}</option>
    );
  };

  const selectMenuItems = (options: SelectOption[], values: string[]) => {
    return options.map(item => (
      <MenuItem key={item.value} value={item.value}>
        <Checkbox checked={values.indexOf(item.value) > -1} />
        <ListItemText primary={item.label} />
      </MenuItem>
    ));
  };

  const getLabels = (selected: string[], labels: Object) => {
    const labelsArray: string[] = [];
    selected.forEach(item =>
      labelsArray.push((labels as any)[item])
    );
    return labelsArray.join(', ');
  };

  return (
    <div className="hide-on-mobile">
      <Grid container={true} spacing={0}>
        <Grid item={true} xs={3}>
          <FormControl fullWidth={true}>
            <InputLabel shrink={true}>
              {InstructorFilterComponent.FieldLabels.Sort}
            </InputLabel>
            <Select
              native={true}
              value={props.sortBy}
              onChange={props.handleChange}
              inputProps={{
                id: InstructorFilterComponent.Ids.Sort,
                name: InstructorFilterComponent.FieldNames.Sort
              }}
            >
              <option value="" disabled={true}>
                {InstructorFilterComponent.Placeholders.Sort}
              </option>
              {selectOptions(sortByOptions)}
            </Select>
          </FormControl>
        </Grid>
        <Grid item={true} xs={3}>
          <FormControl fullWidth={true}>
            <InputLabel shrink={true}>
              {InstructorFilterComponent.FieldLabels.Distance}
            </InputLabel>
            <Select
              native={true}
              value={props.distance}
              onChange={props.handleChange}
              inputProps={{
                id: InstructorFilterComponent.Ids.Distance,
                name: InstructorFilterComponent.FieldNames.Distance
              }}
            >
              <option value="" disabled={true}>
                {placeholder}
              </option>
              {milesSelectItems()}
            </Select>
          </FormControl>
        </Grid>
        <Grid item={true} xs={3}>
          <FormControl fullWidth={true}>
            <InputLabel shrink={true}>
              {InstructorFilterComponent.FieldLabels.Location}
            </InputLabel>
            <Select
              multiple={true}
              value={props.placeForLessons}
              onChange={props.handleChange}
              input={<Input />}
              displayEmpty={true}
              name={InstructorFilterComponent.FieldNames.Location}
              renderValue={(selected: string[]) => {
                if (selected.length === 0) {
                  return InstructorFilterComponent.Placeholders.Location;
                }
                return getLabels(selected, placeForLessonsLabels);
              }}
            >
              {selectMenuItems(placeForLessonsOptions, props.placeForLessons)}
            </Select>
          </FormControl>
        </Grid>
        <Grid item={true} xs={3}>
          <FormControl fullWidth={true}>
            <InputLabel shrink={true}>
              {InstructorFilterComponent.FieldLabels.Availability}
            </InputLabel>
            <Select
              id="demo-mutiple-checkbox"
              multiple={true}
              value={props.availability}
              onChange={props.handleChange}
              input={<Input />}
              displayEmpty={true}
              name={InstructorFilterComponent.FieldNames.Availability}
              renderValue={(selected: string[]) => {
                if (selected.length === 0) {
                  return InstructorFilterComponent.Placeholders.Availability;
                }
                return getLabels(selected, availabilityLabels);
              }}
            >
              {selectMenuItems(availabilityOptions, props.availability)}
            </Select>
          </FormControl>
        </Grid>
        <Grid item={true} xs={3}>
          <FormControl fullWidth={true}>
            <InputLabel shrink={true}>
              {InstructorFilterComponent.FieldLabels.Price}
            </InputLabel>
            <Select
              value={props.priceRange}
              multiple={true}
              input={<Input />}
              displayEmpty={true}
              renderValue={(selected: string[]) => {
                if (selected.length === 0) {
                  return InstructorFilterComponent.Placeholders.Price;
                }
                return `$${selected.join('-')}`;
              }}
            >
              <MenuItem className="price-filter">
                <Slider
                  value={props.priceRange}
                  onChange={props.handlePriceChange}
                  onChangeCommitted={props.handlePriceCommitted}
                  name={InstructorFilterComponent.FieldNames.Price}
                  valueLabelDisplay="auto"
                  marks={InstructorFilterComponent.sliderMarks}
                  step={10}
                  min={InstructorFilterComponent.rangeMinValue}
                  max={InstructorFilterComponent.rangeMaxValue}
                  defaultValue={InstructorFilterComponent.rangeDefaulValue}
                />
              </MenuItem>
            </Select>
          </FormControl>
        </Grid>
        <Grid item={true} xs={3}>
          <FormControl fullWidth={true}>
            <InputLabel shrink={true}>
              {InstructorFilterComponent.FieldLabels.Age}
            </InputLabel>
            <Select
              native={true}
              value={props.age}
              onChange={props.handleChange}
              inputProps={{
                id: InstructorFilterComponent.Ids.Age,
                name: InstructorFilterComponent.FieldNames.Age
              }}
            >
              <option value="" disabled={true}>
                {InstructorFilterComponent.Placeholders.Age}
              </option>
              {selectOptions(ageOptions)}
            </Select>
          </FormControl>
        </Grid>
        <Grid item={true} xs={3}>
          <FormControl fullWidth={true}>
            <InputLabel shrink={true}>
              {InstructorFilterComponent.FieldLabels.Gender}
            </InputLabel>
            <Select
              native={true}
              value={props.gender}
              onChange={props.handleChange}
              inputProps={{
                id: InstructorFilterComponent.Ids.Gender,
                name: InstructorFilterComponent.FieldNames.Gender
              }}
            >
              <option value="" disabled={true}>
                {InstructorFilterComponent.Placeholders.Gender}
              </option>
              {selectOptions(genderOptions)}
            </Select>
          </FormControl>
        </Grid>
        <Grid item={true} xs={3}>
          <FormControl fullWidth={true}>
            <InputLabel shrink={true}>
              {InstructorFilterComponent.FieldLabels.Qualifications}
            </InputLabel>
            <Select
              multiple={true}
              value={props.qualifications}
              onChange={props.handleChange}
              input={<Input />}
              name={InstructorFilterComponent.FieldNames.Qualifications}
              displayEmpty={true}
              renderValue={(selected: string[]) => {
                if (selected.length === 0) {
                  return InstructorFilterComponent.Placeholders.Qualifications;
                }
                return getLabels(selected, qualificationTypes);
              }}
            >
              {selectMenuItems(qualificationOptions, props.qualifications)}
            </Select>
          </FormControl>
        </Grid>
      </Grid>
    </div>
  );
};

export default InstructorFilter;
