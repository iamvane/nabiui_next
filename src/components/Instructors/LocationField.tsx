import * as React from 'react';
import PlacesAutocomplete, {
  geocodeByAddress,
  getLatLng,
} from 'react-places-autocomplete';
import {
  IconButton,
  InputAdornment,
  TextField,
  Typography
} from '@material-ui/core';

import { ListTemplateComponent } from '../common/constants/ListTemplate';

interface Props {
  getLatLng: (lat: string, lng: string) => void;
  getLocation: (location: string) => void;
  address: string;
  getLocationError?: (error: string) => void;
  error?: string;
}

interface State {
  location: string;
  lat: string;
  lng: string;
  error: string;
}

export class LocationField extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      location: '',
      lat: '',
      lng: '',
      error: ''
    };
  }

  public componentDidMount(): void {
    if (this.props.address) {
      geocodeByAddress(this.props.address)
        .then(results => getLatLng(results[0]))
        .then(coordinates => {
          this.setState({
            ...this.state,
            lat: String(coordinates.lat),
            lng: String(coordinates.lng)
          });
          this.props.getLatLng(String(coordinates.lat), String(coordinates.lng));
          this.props.getLocation(this.state.location);
        });
      this.setState({
        location: this.props.address
      });
    }
    this.setState({
      error: this.props.error
    });
  }

  public componentDidUpdate(prevProps: Props): void {
    if (prevProps.address !== this.props.address) {
      this.setState({
        location: this.props.address
      });
    }

    if (prevProps.error !== this.props.error) {
      this.setState({
        error: this.props.error
      })
    }
  }

  public handleLocationChange = (location: string) => {
    this.setState({ location, error: '' });
  }

  public handleLocationSelect = (location: string) => {
    this.setState({ location }, () => {
      geocodeByAddress(location)
        .then(results => getLatLng(results[0]))
        .then(coordinates => {
          this.setState({
            ...this.state,
            lat: String(coordinates.lat),
            lng: String(coordinates.lng)
          });
          this.props.getLatLng(String(coordinates.lat), String(coordinates.lng));
          this.props.getLocation(this.state.location);
        })
        .catch(error => {
          this.props.getLocation('');
          this.setState({ error: 'Enter a valid location.' })
          this.props.getLocationError && this.props.getLocationError('Enter a valid location.')
          console.log('Error', error)
        });
    });
  }

  public render() {
    const locationIcon = 'https://nabimusic.s3.us-east-2.amazonaws.com/assets/images/pin-location.png';
    return (
      <div>
        <PlacesAutocomplete
          value={this.state.location}
          onChange={this.handleLocationChange}
          onSelect={this.handleLocationSelect}
        >
          {({ getInputProps, suggestions, getSuggestionItemProps, loading }) => (
            <div className="nabi-full-width">
              <TextField
                {...getInputProps({
                  variant: "standard",
                  fullWidth: true,
                  placeholder: ListTemplateComponent.locationPlaceholder,
                  InputProps: {
                    startAdornment: (
                      <InputAdornment position="start">
                        <IconButton
                          aria-label="location"
                        >
                          <img src={locationIcon} className="nabi-custom-button-icon" alt="location-icon" />
                        </IconButton>
                      </InputAdornment>
                    ),
                  }
                })}
              />
              <div className="nabi-z-index-1 nabi-position-absolute">
                {loading && <div>Loading...</div>}
                {suggestions.map(suggestion => {
                  const className = suggestion.active
                    ? 'suggestion-item--active'
                    : 'suggestion-item';
                  // inline style for demonstration purpose
                  const style = suggestion.active
                    ? { backgroundColor: '#fafafa', cursor: 'pointer' }
                    : { backgroundColor: '#ffffff', cursor: 'pointer' };
                  return (
                    <div
                      key={suggestion}
                      {...getSuggestionItemProps(suggestion, {
                        className,
                        style,
                      })}
                    >
                      <span>{suggestion.description}</span>
                    </div>
                  );
                })}
              </div>
            </div>
          )}
        </PlacesAutocomplete>
        {this.state.error && <Typography color="error" className="nabi-margin-top-xsmall">{this.state.error}</Typography>}
      </div>
    );
  }
}

export default LocationField;
