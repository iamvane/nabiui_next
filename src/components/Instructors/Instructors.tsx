import * as React from 'react';
import { Redirect } from 'react-router-dom';

import InstructorCard from 'components/Instructors/InstructorCard';
import { Instructor, Rates, InstructorType } from 'redux/models/InstructorModel';
import { Route } from 'components/Instructors/constants';

interface Props {
  instructors: Instructor[];
  isRequestingInstructor: boolean;
  instructor: InstructorType;
  fetchInstructor: (id: number) => void;
}

const Instructors: React.StatelessComponent<Props> = props => {
  return (
    <div>
      {
        props.instructor.id ? <Redirect to={`${Route.Profile}/${props.instructor.id}`} /> :
        <React.Fragment>
          {props.instructors.length > 0 && props.instructors.map((instructor: Instructor, i: number) => (
            <InstructorCard
              key={i}
              id={instructor.id as number}
              fetchInstructor={props.fetchInstructor}
              age={instructor.age ? instructor.age : 0}
              memberSince={instructor.memberSince ? instructor.memberSince : ''}
              lastLogin={instructor.lastLogin ? instructor.lastLogin : ''}
              address={instructor.address ? instructor.address : ''}
              displayName={instructor.displayName ? instructor.displayName : ''}
              reviewsNumber={instructor.reviews ? instructor.reviews : 0}
              instruments={instructor.instruments ? instructor.instruments as any : []}
              backgroundCheck={instructor.backgroundCheck ? instructor.backgroundCheck : false}
              favorite={instructor.favorite ? instructor.favorite : false}
              lessonsTaught={instructor.lessonsTaught ? instructor.lessonsTaught : 0}
              avatarImage={instructor.avatar as string}
              rateStartAt={instructor.rates as Rates}
              bioTitle={instructor.bioTitle as string}
              bioDescription={instructor.bioDescription as string}
            />
          ))}
        </React.Fragment>}
    </div>
  );
};

export default Instructors;
