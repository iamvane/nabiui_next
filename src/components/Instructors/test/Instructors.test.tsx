import * as React from 'react';
import { shallow } from 'enzyme';

import Instructors from '../Instructors';

describe('Instructors', () => {
  const  wrapper = shallow(
    <Instructors
      instructors={[{}]}
      fetchInstructor={jest.fn()}
      isRequestingInstructor={false}
      instructor={{
        userId: 0
      }}
    />
  );

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
