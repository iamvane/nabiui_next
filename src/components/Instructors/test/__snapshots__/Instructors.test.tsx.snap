// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructors Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Instructors
    fetchInstructor={[MockFunction]}
    instructor={
      Object {
        "userId": 0,
      }
    }
    instructors={
      Array [
        Object {},
      ]
    }
    isRequestingInstructor={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <UNDEFINED>
        <InstructorCard
          address=""
          age={0}
          avatarImage={undefined}
          backgroundCheck={false}
          bioDescription={undefined}
          bioTitle={undefined}
          displayName=""
          favorite={false}
          fetchInstructor={[MockFunction]}
          id={undefined}
          instruments={Array []}
          lastLogin=""
          lessonsTaught={0}
          memberSince=""
          rateStartAt={undefined}
          reviewsNumber={0}
        />
      </UNDEFINED>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <InstructorCard
            address=""
            age={0}
            avatarImage={undefined}
            backgroundCheck={false}
            bioDescription={undefined}
            bioTitle={undefined}
            displayName=""
            favorite={false}
            fetchInstructor={[MockFunction]}
            id={undefined}
            instruments={Array []}
            lastLogin=""
            lessonsTaught={0}
            memberSince=""
            rateStartAt={undefined}
            reviewsNumber={0}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "address": "",
            "age": 0,
            "avatarImage": undefined,
            "backgroundCheck": false,
            "bioDescription": undefined,
            "bioTitle": undefined,
            "displayName": "",
            "favorite": false,
            "fetchInstructor": [MockFunction],
            "id": undefined,
            "instruments": Array [],
            "lastLogin": "",
            "lessonsTaught": 0,
            "memberSince": "",
            "rateStartAt": undefined,
            "reviewsNumber": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <UNDEFINED>
          <InstructorCard
            address=""
            age={0}
            avatarImage={undefined}
            backgroundCheck={false}
            bioDescription={undefined}
            bioTitle={undefined}
            displayName=""
            favorite={false}
            fetchInstructor={[MockFunction]}
            id={undefined}
            instruments={Array []}
            lastLogin=""
            lessonsTaught={0}
            memberSince=""
            rateStartAt={undefined}
            reviewsNumber={0}
          />
        </UNDEFINED>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <InstructorCard
              address=""
              age={0}
              avatarImage={undefined}
              backgroundCheck={false}
              bioDescription={undefined}
              bioTitle={undefined}
              displayName=""
              favorite={false}
              fetchInstructor={[MockFunction]}
              id={undefined}
              instruments={Array []}
              lastLogin=""
              lessonsTaught={0}
              memberSince=""
              rateStartAt={undefined}
              reviewsNumber={0}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "address": "",
              "age": 0,
              "avatarImage": undefined,
              "backgroundCheck": false,
              "bioDescription": undefined,
              "bioTitle": undefined,
              "displayName": "",
              "favorite": false,
              "fetchInstructor": [MockFunction],
              "id": undefined,
              "instruments": Array [],
              "lastLogin": "",
              "lessonsTaught": 0,
              "memberSince": "",
              "rateStartAt": undefined,
              "reviewsNumber": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
